### PR TITLE
support specify partId mode

### PR DESCRIPTION
**Pull request by panda-sheep**
#### What type of PR is this?
- [ ] bug
- [X] feature
- [ ] enhancement

#### What does this PR do?
 support specify partId mode in upgrade tool

#### Checklist：
- [ ] Documentation affected （Please add the label if documentation needs to be modified.)
- [ ] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [ ] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:
>                                                                 `

### CODE REVIEW

File: src/tools/db-upgrade/DbUpgraderTool.cpp [hunk 1]
```diff
@@ -164,15 +174,96 @@ int main(int argc, char* argv[]) {
...
+    auto space = subDirs[0];
+    GraphSpaceID spaceId;
+    try {
+      spaceId = folly::to<GraphSpaceID>(space);
+    } catch (const std::exception& ex) {
+      LOG(ERROR) << "Cannot convert space id " << space;
+      return EXIT_FAILURE;
+    }
+
+    auto sRet = schemaMan->toGraphSpaceName(spaceId);
```
Comment (critical27) [inline src/tools/db-upgrade/DbUpgraderTool.cpp:218]: spaceName is useless?

File: src/tools/db-upgrade/DbUpgraderTool.cpp [hunk 2]
```diff
@@ -164,15 +174,96 @@ int main(int argc, char* argv[]) {
...
+      }
+      iter->next();
+    }
+  }
+  if (!partsAll.empty()) {
+    std::string partstr;
+    for (auto part : partsAll) {
+      partstr += folly::stringPrintf("%d ", part);
+    }
+    LOG(FATAL) << "There are some part " << partstr << "not found";
```
Comment (critical27) [inline src/tools/db-upgrade/DbUpgraderTool.cpp:260]: Good! Just use `LOG(ERROR)` and `return EXIT_FAILURE;`, no need to core.

File: src/tools/db-upgrade/DbUpgrader.cpp [hunk 3]
```diff
@@ -322,33 +403,254 @@ void UpgraderSpace::runPartV1() {
...
         lastSrcVertexId = svId;
         lastEdgeType = edgetype;
         lastRank = ranking;
         lastDstVertexId = dvId;
       }
 
-      if (data.size() >= FLAGS_write_batch_num) {
-        VLOG(2) << "Send record total rows " << data.size();
+      // tag data
+      if (tagData.size() >= FLAGS_write_batch_num) {
```
Comment (critical27) [inline src/tools/db-upgrade/DbUpgrader.cpp:422]: Some code could be unified, here and below when `if (tagData.size() > 0) {`, but it is ok for this tool.

**Review by critical27 — COMMENTED** [review]
Good job, thx a lot.

File: src/tools/db-upgrade/DbUpgrader.cpp (hunk 3 above)
Comment (panda-sheep) [reply src/tools/db-upgrade/DbUpgrader.cpp:482]: yeah, fixed.

File: src/tools/db-upgrade/DbUpgraderTool.cpp (hunk 1 above)
Comment (panda-sheep) [reply src/tools/db-upgrade/DbUpgraderTool.cpp:218]: Here only check whether the space exists or not

File: src/tools/db-upgrade/DbUpgraderTool.cpp (hunk 2 above)
Comment (panda-sheep) [reply src/tools/db-upgrade/DbUpgraderTool.cpp]: ok

**Pull request merged**